test/exandra/integration_test.exs

```diff
@@ -17,15 +22,7 @@ defmodule Exandra.IntegrationTest do
       "CREATE KEYSPACE IF NOT EXISTS exandra_integration WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };"
     )
 
-    previous_adapter = Application.fetch_env!(:exandra, :adapter)
-    previous_child_spec = Application.fetch_env!(:exandra, :child_spec)
-    Application.put_env(:exandra, :adapter, Exandra.Adapter.XandraClustered)
-    Application.delete_env(:exandra, :child_spec)
-
-    on_exit(fn ->
-      Application.put_env(:exandra, :adapter, previous_adapter)
-      Application.put_env(:exandra, :child_spec, previous_child_spec)
-    end)
+    stub_with(XandraClusterMock, Exandra.XandraCluster)
```

Comment:
Super-duper nice Mox trick to read the adapter at compile time, and then still fall back to the real adapter for integration tests